source/sv_main.c

```diff
@@ -1473,48 +1469,47 @@ void Load_Waypoint ()
 		}
 	}
 	Con_DPrintf("Total waypoints: %i\n", i);
-	for (i = 1;i < MAX_WAYPOINTS; i++) //for sake of saving time later we are now going to save each targets array position and distace to each waypoint
+	for (i = 0; i < MAX_WAYPOINTS; i++) //for sake of saving time later we are now going to save each targets array position and distace to each waypoint
 	{
-		for (p = 0;waypoints[i].target[p]; p++)
+		for (p = 0; waypoints[i].target[p]; p++)
 		{
-			for (s = 1; s < MAX_WAYPOINTS; s++)
+			if (waypoints[i].target[p] < 0) break;
+	
+			for (s = 0; s < MAX_WAYPOINTS; s++)
 			{
-				if (s == MAX_WAYPOINTS)
-					Sys_Error ("Waypoint (%i) without a target!\n", s);
-				if (waypoints[i].target[p] == waypoints[s].id)
+				if (waypoints[i].target[p] == s)
 				{
 					waypoints[i].dist[p] = VecLength2(waypoints[s].origin, waypoints[i].origin);
-					waypoints[i].target_id[p] = s;
 					break;
 				}
 			}
 		}
-			Con_DPrintf("Waypoint (%i)target: %i (%i, %f), target2: %i (%i, %f), target3: %i (%i, %f), target4: %i (%i, %f), target5: %i (%i, %f), target6: %i (%i, %f), target7: %i (%i, %f), target8: %i (%i, %f)\n",
-			waypoints[i].id,
-			waypoints[i].target[0],
-			waypoints[i].target_id[0],
-			waypoints[i].dist[0],
-			waypoints[i].target[1],
-			waypoints[i].target_id[1],
-			waypoints[i].dist[1],
-			waypoints[i].target[2],
-			waypoints[i].target_id[2],
-			waypoints[i].dist[2],
-			waypoints[i].target[3],
-			waypoints[i].target_id[3],
-			waypoints[i].dist[3],
-			waypoints[i].target[4],
-			waypoints[i].target_id[4],
-			waypoints[i].dist[4],
-			waypoints[i].target[5],
-			waypoints[i].target_id[5],
-			waypoints[i].dist[5],
-			waypoints[i].target[6],
-			waypoints[i].target_id[6],
-			waypoints[i].dist[6],
-			waypoints[i].target[7],
-			waypoints[i].target_id[7],
-			waypoints[i].dist[7]);
+		Con_Printf("Waypoint (%i)\n target: %i (%i, %f),\n target2: %i (%i, %f),\n target3: %i (%i, %f),\n target4: %i (%i, %f),\n target5: %i (%i, %f),\n target6: %i (%i, %f),\n target7: %i (%i, %f),\n target8: %i (%i, %f)\n",
```

Comment:
Please make this a developer print instead to avoid increase in load time (+ console/log spam).